Suggest updated version of git-pr

lagru opened the pull request:
Here's my own version of your script. Take it or leave it. :wink:

lagru committed:
Suggest updated version with cleanup

stefanv commented:
This looks like a nice improvement, thanks! How about also setting up the remote branch, so `git push` "just works"?

lagru commented:
It's already setup that way. However, git complains by default, that the remote and local branch names are different. That's because this script adds `pr/<NUMBER>_` to the local branch name.

stefanv commented:
Cool, let's get the remote names normalized, then I'll go ahead and merge. Note that I added a BSD0 license to the repo, to make it maximally accessible.

lagru committed:
Add push command

Also explain the current action with a blue comment and use clearer
variable names for branch and remote.

lagru commented:
I'm now using the prefix `ghpr_` for remotes and `ghpr/` for branches to make it clear that either are managed by this tool. I assume you don't like those prefixes either?

This also got me thinking, do we need to worry about malicious remote or branch names here? :thinking:

lagru commented on scripts/git-pr:
```diff
@@ -1,38 +1,227 @@
+        check=check,
+        text=True,
+        stdout=subprocess.PIPE,
+        # Never use shell=True here to prevent shell injections,
+        # as args might contain input fetched from the web
+        shell=False,
```
Never mind the question about shell injection. Forgot the script had this.

lagru commented on scripts/git-pr:
```diff
@@ -1,38 +1,227 @@
+        metavar="NUMBER",
+        help="Number of the pull request to check out",
+    )
+    parser.add_argument(
+        "-p",
+        "--push",
+        action="store_true",
+        help="Push changes on PR branch to remote (assume existing PR branch)",
```
I added the `--push` option as a helper for users that haven't configured their git to not complain about differing local and remote branch names, which is probably most users.

Couldn't think of a better option and this works for me perfectly so far. :shrug:
